src/process_tree.rs

```diff
@@ -0,0 +1,102 @@
+// SPDX-FileCopyrightText: 2025 blinry <mail@blinry.org>
+//
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+use std::{ops::Deref, path::PathBuf};
+
+use procfs::{process::Process, ProcError, ProcResult};
+
+#[derive(Debug)]
+pub struct CwdProcessTree {
+    tree: ProcessTree,
+    cwd: PathBuf,
+}
+
+#[derive(Debug)]
+pub struct ProcessTree {
```

Comment:
Maybe this should get a new name, now that it's no longer a tree :D Just a [newtype](https://www.howtocodeit.com/articles/ultimate-guide-rust-newtypes) for `procfs::process::Process` called `Process`, I guess?

And instead of a separate `CwdProcessTree`, could the new `Process` type get a `cwd()` method?